shared/react-native/ios/KeybaseShare/FilesViewController.m

```diff
@@ -0,0 +1,222 @@
+                              [NSNumber numberWithInt:1], @"direntType",
+                              nil]];
+    [itemsWithBack addObjectsFromArray:sortedAndFilteredItems];
+    [self setDirectoryEntries:itemsWithBack];
+  }
+}
+
+- (void)didReceiveMemoryWarning {
+  KeybaseExtensionForceGC();
+  [super didReceiveMemoryWarning];
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+  return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  return [self.directoryEntries count];
+}
+
+- (NSDictionary*)getItemAtIndex:(NSIndexPath*)indexPath {
+  NSInteger index = [indexPath item];
+  return self.directoryEntries[index];
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+  UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"ConvCell"];
+  if (NULL == cell) {
+    cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"ConvCell"];
+  }
+  NSDictionary* item = [self getItemAtIndex:indexPath];
+  [[cell textLabel] setText:item[@"name"]];
+  if ([[item objectForKey:@"direntType"] intValue] == 1) {
+    // Style folders differently from files. Use Keybase blue.
+    [[cell textLabel] setTextColor:[UIColor colorWithRed:76.0/255.0 green:142.0/255.0 blue:1 alpha:1]];
+  } else {
+    [[cell textLabel] setTextColor:[UIColor colorWithRed:0 green:0 blue:0 alpha:1]];
+  }
+  return cell;
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+  NSDictionary* target = [self getItemAtIndex:indexPath];
+  if ([indexPath isEqual:[NSIndexPath indexPathForRow:0 inSection:0]] && [[target objectForKey:@"name"] isEqualToString:UpOneLevel]) {
+    // '..' navigates back up.
+    NSArray* pathElems = [self.path componentsSeparatedByString:@"/"];
+    NSArray* upOneDirectory = [pathElems subarrayWithRange:NSMakeRange(0, [pathElems count] - 2)];
+    [self setPath:[NSString stringWithFormat:@"%@/", [upOneDirectory componentsJoinedByString:@"/"]]];
+    [self dispatchFilesBrowser];
+  } else if ([[target objectForKey:@"direntType"] intValue] == 1) {
+    // Folders navigate down.
+    [self setPath:[NSString stringWithFormat:@"%@%@/", self.path, target[@"name"]]];
+    [self dispatchFilesBrowser];
+  } else {
+    // Files do not affect navigation.
+    [tableView deselectRowAtIndexPath:indexPath animated:FALSE];
+  }
+}
+
+- (void)sendPathToDelegate {
+  [self.delegate folderSelected:self.path];
+}
+
+- (unsigned long)pathLength {
+  NSArray* pathElems = [self.path componentsSeparatedByString:@"/"];
+  return [pathElems count];
+}
+
+- (void)setPath:(NSString*)path {
+  _path = path;
+  NSArray* pathElems = [self.path componentsSeparatedByString:@"/"];
+  unsigned long pathLength = [pathElems count];
+  if (pathLength <= 2) {
+    [self setTitle:@"Keybase"];
```

Comment:
Do we need `self.navigationItem.rightBarButtonItem = nil;` here too?

Reply:
Yep, good call.